### PR TITLE
Issue #404: Build with and target SDK 28 (+ update support libraries).

### PR DESCRIPTION
**Pull request by pocmo**
Cherry-picking the SDK 28 patch to re-apply.

### CODE REVIEW

**Comment by pocmo** [comment]
This should be ready to land now. I rebased the PR and fixed some small things. Let's see what taskcluster says.

**Comment by pocmo** [comment]
The decision task is failing for some reason. I'll try to debug this.

**Comment by pocmo** [comment]
Apart from the build issue (#1337) some new tests need to be fixed.

**Comment by pocmo** [comment]
Okay, this should pass now. :)

File: components/browser/menu/src/test/java/mozilla/components/browser/menu/BrowserMenuBuilderTest.kt [hunk 1]
```diff
@@ -28,7 +28,7 @@ class BrowserMenuBuilderTest {
         val recyclerView: RecyclerView = popup.contentView.findViewById(R.id.mozac_browser_menu_recyclerView)
         assertNotNull(recyclerView)
 
-        val recyclerAdapter = recyclerView.adapter
+        val recyclerAdapter = recyclerView.adapter!!
```
Comment (Amejia481) [inline components/browser/menu/src/test/java/mozilla/components/browser/menu/BrowserMenuBuilderTest.kt]: What's the difference to change it **!!**? Just to be more explicit about nullability?

File: components/service/telemetry/src/test/java/org/mozilla/telemetry/measurement/OperatingSystemVersionMeasurementTest.java [hunk 2]
```diff
@@ -25,7 +25,7 @@ public void testValue() {
 
         final String version = (String) value;
         assertFalse(TextUtils.isEmpty(version));
-        assertEquals("27", version); // 16 is the default API level of this Robolectric version
+        assertEquals("28", version);
```
Comment (Amejia481) [inline components/service/telemetry/src/test/java/org/mozilla/telemetry/measurement/OperatingSystemVersionMeasurementTest.java]: Maybe we can have a constant for that, this way we have only one point of changing

File: components/browser/menu/src/test/java/mozilla/components/browser/menu/BrowserMenuBuilderTest.kt (hunk 1 above)
Comment (pocmo) [reply components/browser/menu/src/test/java/mozilla/components/browser/menu/BrowserMenuBuilderTest.kt]: They added @Nullable in the latest version of the support library. Now we need to deal with that :)

Comment (Amejia481) [reply components/browser/menu/src/test/java/mozilla/components/browser/menu/BrowserMenuBuilderTest.kt]: 👍

**Pull request merged**